articles/3_sass-selectors.md

```diff
@@ -0,0 +1,237 @@
+# A Crash Course in Selectors: Part 3
+
+Up until this point, I have only been covering features that can be used in pure, out of the box CSS. While CSS can do a lot on its own, there are a bunch of really powerful pre-processors that compile into CSS that allow developers to iterate faster, build more complex systems with less code, and save a lot of time and energy.
+
+At Addepar, we use [SASS](http://sass-lang.com/), a pre-processor "with superpowers", to borrow their language. SASS has a ton of awesome features, most of which deserve their own blog post. In the spirit of continuing our conversation on selectors, I thought I'd highlight the ways you can play around with selectors using SCSS syntax.
+
+## A note about SASS vs SCSS
+A lot of the time you will see SASS files with the file extension `.scss`. This denotes a file with *Sassy CSS* syntax. The difference being that any valid `.css` file can be read by an SCSS compiler and it will produce itself. There are some additional differences between SASS and SCSS, the largest of which being that SASS does not require curly braces, instead using tabs (similar to Python).
+
+For this post (and all following, unless I specify otherwise), I will be using SCSS syntax. If you're interested in learning more about the pro's and con's of each syntax, read this article [here](http://thesassway.com/editorial/sass-vs-scss-which-syntax-is-better).
+
+## Nesting
+
+One of the most annoying parts about writing vanilla CSS is that it's very hard to keep organized without tons of documentation, especially in a large system. A big part of why this is messy, is that its hard to tell which elements are children of which parents, so elements will inherit unexpected styling thats difficult to debug.
+
+In SASS, you can nest selectors like this:
+
+```scss
+.parent {
+  /* styling goes here */
+
+  .descendant {
+    /* descendant-specific styles go here */
+  }
+}
+```
+
+And it will get compiled into CSS that looks like this:
+
+```css
+.parent {
+  /* styling goes here */
+}
+
+.parent .descendant {
+  /* descendant-specific styles go here */
+}
+```
+
+Note that I'm using `descendant` instead of `child`, because this nesting works for any descendant of the parent, not just its direct children.
+
+For added specificity, you can also use this nesting with any parent and any descendant selectors (see [A Crash Course in CSS Selectors: Part 1](1_selectors.md) if you need a refresher).
+
+
+## The Ampersand
+
+If you want to get more ambitious than just nesting children from parents, you can use an ampersand (`&`) as a concatenator within nested SASS structures. There are some other idiosyncrasies with the `&`, so we will go into each of them with more detail.
```

Comment:
I'd add an explanation for what the `&` actually is, which is basically a variable for the previous selector. The `&` also inherits all of that parent's parents.
Not sure if this is more advanced than you wanted to get, but:
```
.parent {
    .child {
       & .baby {}
    }
}
```
People (as in me in the past lol) might think the `&` in this case is `.child` but its actually `.parent .child`. This becomes important when you use the `&` in this way:
```
.parent {
    .child {
       .foo & {}
    }
}
```
You might expect it to compile to `.parent .foo .child`
But it actually compiles to `.foo .parent .child` which doesn't work. So I always make sure there is only one level of nesting happening when I do that trick.